packages/fuels-accounts/src/lib.rs

```diff
@@ -406,11 +408,14 @@ mod tests {
         .into();
 
         // Sign the transaction.
+        let consensus_parameters = ConsensusParameters::default();
+        let test_provider = Provider::new(FuelClient::new("test")?, consensus_parameters);
+        wallet.set_provider(test_provider);
```

Comment:
would it make sense to use the default `ConsensusParameters` when there is no provider and the wallet is signing ? Then we would not have to make a dummy provider just to sign a tx.

The `sign_transaction` would not return a result (it would not fail) but instead just a `Signature` made with the default values.

Reply:
I had already suggested some nice usability defaults but they were dismissed. If you think the UX can be improved, it will be done in a separate PR, I will not trigger another review cycle.